src/Aperture/elliptical.jl

```diff
@@ -0,0 +1,89 @@
+export EllipticalAperture,
+       EllipticalAnnulus
+
+"""
+   EllipticalAperture(x, y, a, b, theta)
+
+A elliptical aperture.
+
+Where a >= b > 0 and -pi/2 <= theta <= pi/2
+
+# Examples
+```jldoctest
+julia> EllipticalAperture(2,2,2,2,1.2)
+EllipticalAperture(2.0, 2.0, a=2.0, b=2.0, theta=1.2)
+
+```
+"""
+struct EllipticalAperture{T <: Number} <: AbstractAperture
+    x::T
+    y::T
+    a::T
+    b::T
+    theta::T
+    cxx::T
+    cxy::T
+    cyy::T
+end
+
+function EllipticalAperture(x, y, a, b, theta)
+    if (b < 0.0 || a < b || theta < -pi/2 || theta > pi/2)
+            error("illegal ellipse parameters. " * "Require a >= b > 0.0, -pi/2 <= theta <= pi/2")
+    end
+    cxx, cyy, cxy = oblique_coefficients(a, b, theta)
+    return EllipticalAperture(promote(x, y, a, b, theta, cxx, cxy, cyy)...)
+
+end
+
+function Base.show(io::IO, e::EllipticalAperture)
+    print(io, "EllipticalAperture($(e.x), $(e.y), a=$(e.a), b=$(e.b), theta=$(e.theta))")
+end
+
+function oblique_coefficients(a, b, theta)
+    sintheta, costheta = sincos(theta)
+    asq = a*a
+    bsq = b*b
+    cxx = costheta*costheta*inv(asq) + sintheta*sintheta*inv(bsq)
+    cyy = sintheta*sintheta*inv(asq) + costheta*costheta*inv(bsq)
+    cxy = 2*costheta*sintheta * (inv(asq) - inv(bsq))
+    return cxx, cyy, cxy
```

Comment:
It makes more sense to use `a^2` instead of `a*a` and `_ / asq` instead of `* inv(asq)` for readability.
```julia
sint, cost = sincos(deg2rad(theta))
a2 = a^2
b2 = b^2
cxx = cost^2 / a2 + sint^2 / b2
cyy = sint^2 / a2 + cost^2 / b2
czz = 2cost * sint * (1/a2 - 1/b2)
```

Reply:
https://github.com/JuliaAstro/Photometry.jl/pull/4#discussion_r375413368

Mosè suggested using inv instead of 1/__ in the above-mentioned discussion.

Reply:
Yes, I suggested `inv(x)` because for some types it's more efficient than `1 / x`, but I'll let @mileslucas choose :slightly_smiling_face:

Reply:
Oh, I did not know that! In that case it doesn't matter to me which you use, as long as it is readable and clear what is happening.